### PR TITLE
Add Eats native SSO authentication option; fallback ability between native apps

### PR DESCRIPTION
**Pull request by ebgraham**
The new parameter `productFlowPriority` on LoginManager allows you to authenticate with UberEats or Uber rides, with the ability to fallback between them (e.g. if Eats isn't installed then try the rides app).  By default, uses Uber rides app which makes this a non-breaking change.

Right now for web auth UberEats will still use the rides flow, but in a future update it will use its own web auth provider.

Due to Obj-c compatibility the API is a bit messier than I would've preferred.

### CODE REVIEW

File: source/UberCore/Authentication/LoginManager.swift [hunk 1]
```diff
@@ -40,16 +41,18 @@ import SafariServices
...
-    - parameter loginType:         The login type to use for logging in, defaults to Implicit
-    
+    - parameter keychainAccessGroup: The keychain access group to use for saving the Access Token, defaults to Configuration.shared.defaultKeychainAccessGroup
+    - parameter loginType: The login type to use for logging in, defaults to Implicit
+    - parameter productFlowPriority: An ordered list of the Uber apps to use for authentication, if available. Defaults to trying to use only the main Uber rides flow.
+
     - returns: An initialized LoginManager
     */
-    @objc public init(accessTokenIdentifier: String, keychainAccessGroup: String?, loginType: LoginType) {
+    @objc public init(accessTokenIdentifier: String, keychainAccessGroup: String?, loginType: LoginType, productFlowPriority: [UberAuthenticationProductFlow] = [UberAuthenticationProductFlow(.rides)]) {
```
Comment (edjiang) [inline source/UberCore/Authentication/LoginManager.swift]: The reason why we have all these convenience inits is that objc won't see the default values provided by `productFlowPriority: [UberAuthenticationProductFlow] = [UberAuthenticationProductFlow(.rides)]`. So I would suggest not putting default values here, and instead create more convenience inits.

File: source/UberCore/Authentication/LoginManager.swift [hunk 2]
```diff
@@ -261,34 +273,69 @@ import SafariServices
...
+        var stop: Bool = false
+        var fallback: Bool = false
+        
+        for authenticator in authenticators where stop == false {
+            self.authenticator = authenticator
+            fallback = false
+
+            DeeplinkManager.shared.open(authenticator.authorizationURL, completion: { error in
+                if error == nil {
+                    return stop = true
```
Comment (edjiang) [inline source/UberCore/Authentication/LoginManager.swift]: I'm surprised this is valid swift. This callback should have a returntype of void. Also, your code seems to assume that this runs synchronously. I don't think that's a guarantee that is valid.

File: source/UberCoreTests/LoginManagerTests.swift [hunk 3]
```diff
@@ -128,7 +128,7 @@ class LoginManagerTests: XCTestCase {
         let testSourceApplication = "com.ubercab.foo"
         let testAnnotation = "annotation"
 
-        let authenticatorMock = NativeAuthenticatorPartialStub(scopes: [.profile])
+        let authenticatorMock = RidesNativeAuthenticatorPartialStub(scopes: [.profile])
```
Comment (edjiang) [inline source/UberCoreTests/LoginManagerTests.swift]: Should probably add some tests for the Eats deeplinks?

File: README.md [hunk 4]
```diff
@@ -345,7 +345,23 @@ CLLocation *pickupLocation = [[CLLocation alloc] initWithLatitude: 37.787654 lon
...
+The easiest form of authorization is using the `.native` login type. It allows you to request Privileged scopes and, if your user is logged into the Uber app, it doesn't require your user to enter a username and password. It requires the user to have the native Uber application on their device. 
+
+Native authentication can occur either through the main Uber rides app or through the Uber Eats app. You can also fallback from one app to another, so that for example if the user does not have the UberEats app installed on their device, you can instead authenticate through the main Uber rides app:
+
+```swift
+// Swift
+let loginManager = LoginManager(loginType: .native, productFlowPriority: [UberAuthenticationProductFlow(.eats)], [UberAuthenticationProductFlow(.rides)])
```
Comment (edjiang) [inline README.md]: Typo, should be in the same array

**Comment by ebgraham** [comment]
updated this PR, but realized I didn't open it against develop. going to re-open against develop branch

**Pull request closed without merging**